app/gui/src/model/project/synchronized.rs

```diff
@@ -225,6 +225,20 @@ async fn reload_module_on_file_change(
     Ok(())
 }
 
+/// Update internal state when the `refactoring/projectRenamed` notification received.
+#[profile(Detail)]
+fn rename_project_on_notification(
+    project_renamed: language_server::types::ProjectRenamed,
+    properties: Rc<RefCell<Properties>>,
+    execution_contexts: Rc<ExecutionContextsRegistry>,
+) {
+    let mut properties = properties.borrow_mut();
+    properties.displayed_name = project_renamed.new_name.into();
+    properties.project_name.project = project_renamed.new_normalized_name.clone().into();
+    execution_contexts
```

Comment:
This is most likely fine currently, but it might be prudent to explicitly limit the mutable borrow doing another function call to avoid an accidental double borrow somewhere else.

Reply:
@4e6 FYI, you can either add `drop(properties)` after the last line that uses them or surround `properties`-related code with curly braces to do that.